napi/parser/parser.wasi-browser.js

```diff
@@ -5,7 +5,11 @@ import {
   createOnMessage as __wasmCreateOnMessageForFsProxy,
 } from '@napi-rs/wasm-runtime'
 
-import __wasmUrl from './parser.wasm32-wasi.wasm?url'
+let __wasmUrl = new URL('./parser.wasm32-wasi.wasm', import.meta.url).href
+// https://github.com/esm-dev/esm.sh/issues/1109
+if (import.meta.url.includes("esm.sh")) {
+  __wasmUrl = __wasmUrl.replace("/es2022/", "/");
+}
```

Comment:
This one works
https://esm.sh/gh/oxc-project/oxc@1cb56259406aa6b2531b7686982864a22b9f3825/napi/parser/parser.wasi-browser.js